tests/tests/test_monitor_client.py

```diff
@@ -1162,6 +1162,30 @@ def test_monitorclient_logs_and_services(self, monitor_commercial_setup_no_clien
                 + devid
             )
 
+        fds_count_timeout_s = 64
+        max_fds_count_diff = 2
+        logger.info("test_monitorclient_alert_email: checking open file descriptors")
+        time.sleep(fds_count_timeout_s)
+        fds_count0 = mender_device.run(
+            "ls /proc/$(cat /var/run/monitoring-client.pid)/fd | wc -l"
+        )
+        assert fds_count0.rstrip().isnumeric()
+        fds_count0=int(fds_count0.rstrip())
+        logger.info(
+            "test_monitorclient_alert_email: currently %s fds open" % fds_count0
+        )
+        time.sleep(fds_count_timeout_s)
+        fds_count1 = mender_device.run(
+            "ls /proc/$(cat /var/run/monitoring-client.pid)/fd | wc -l"
+        )
+        assert fds_count1.rstrip().isnumeric()
+        fds_count1=int(fds_count1.rstrip())
+        logger.info(
+            "test_monitorclient_alert_email: currently %s fds open" % fds_count0
+        )
+        assert abs(fds_count1 - fds_count0) <= max_fds_count_diff
```

Comment:
Curious: why do we need this tolerance of 2 FD difference?

Reply:
it is possible that at the beginning we caught the daemon before the start of both logging checks, in that case the open fd count is 4. at the end we usually have 6, since both checks add at least one open fd. depending on which "sleep interval" we caught at the last check, the open fd count can fluctuate between these two. but one thing for sure: it never should increase. makes sense?